### PR TITLE
Task 3: React query

### PR DESCRIPTION
**Pull request by kwarnerredhat**

### CODE REVIEW

File: src/pages/Landing/index.tsx [hunk 1]
```diff
@@ -16,12 +16,22 @@ import {
...
 import { useAppContext } from 'src/middleware';
+import ReactDOM from 'react-dom';
+import App from './App';
 
+const queryClient = new QueryClient();
+
+ReactDOM.render(
+  <QueryClientProvider client={queryClient}>
+    <App />
+  </QueryClientProvider>,
```
Comment (Andrewgdewar) [inline src/pages/Landing/index.tsx]: So, this is a "Wrapper component" or "higher order component" (HOC). 
You need to add the queryClientProvider to wherever the top level of the app. 
In our apps case, we already have that done [here](https://github.com/content-services/React-typescript-demo/blob/main/src/index.tsx#L16C5-L16C47). 
Task 3 is asking you to use react-query to add the api calls to the modal component on submit.
IE call the api, with react query, on submit.

File: src/App.tsx [hunk 2]
```diff
@@ -5,6 +5,40 @@ import { customersGetter, customersSetter } from './api/CustomerApi';
...
+import ReactDOM from 'react-dom';
+
+const queryClient = new QueryClient();
+
+ReactDOM.render(
+  <QueryClientProvider client={queryClient}>
+    <App />
+  </QueryClientProvider>,
+  document.getElementById('root')
+);
```
Comment (Andrewgdewar) [inline src/App.tsx]: This shouldn't be needed, it's already implemented [here](https://github.com/content-services/React-typescript-demo/blob/main/src/index.tsx#L16C5-L16C47).

File: src/App.tsx [hunk 3]
```diff
@@ -5,6 +5,40 @@ import { customersGetter, customersSetter } from './api/CustomerApi';
...
+      'Content-Type': 'application/json',
+    },
+  });
+
+  if (!response.ok) {
+    throw new Error('Failed to add customer');
+  }
+
+  return response.json();
+};
```
Comment (Andrewgdewar) [inline src/App.tsx]: This is a standard api call, but to an api that does not exist. 
If you read task 3, it tells you that you have to use the api (that is already completed for you)[here](https://github.com/content-services/React-typescript-demo/blob/main/src/api/CustomerApi.ts#L38).

File: src/pages/Landing/index.tsx [hunk 4]
```diff
@@ -70,9 +63,16 @@ export default () => {
         </Button>
       </GridItem>
       <GridItem sm={6}>
-        <Button onClick={() => setIsModalOpen(true)} variant='secondary'>
+        <Button
+          onClick={() => {
+
+            mutate({});
```
Comment (Andrewgdewar) [inline src/pages/Landing/index.tsx]: Continue working here with the above mentioned API.

File: src/App.tsx [hunk 5]
```diff
@@ -5,6 +5,40 @@ import { customersGetter, customersSetter } from './api/CustomerApi';
...
+const { mutate } = useMutation(addNewCustomer, {
+  onSuccess: () => {
+
+    queryClient.invalidateQueries('customers');
+  },
+});
```
Comment (Andrewgdewar) [inline src/App.tsx]: This is the only portion of code that may in part be used.
But we wouldn't place this in the app.tsx file.

Comment (kwarnerredhat) [reply src/App.tsx]: I placed in the App.tsx because <App ./> gave me an error so i had to imported from app

Comment (kwarnerredhat) [reply src/App.tsx]: Nvm I see what you mean lol

File: src/pages/Landing/index.tsx [hunk 6]
```diff
@@ -60,9 +63,15 @@ export default () => {
         </Button>
       </GridItem>
       <GridItem sm={6}>
-        <Button onClick={() => setIsModalOpen(true)} variant='secondary'>
+        <Button
+
+          onClick={handleAddNewCustomer}
```
Comment (Andrewgdewar) [inline src/pages/Landing/index.tsx:68]: is this imported?
I would expect this code to have compilation errors.

**Pull request closed without merging**